libdevcore/FileSystem.h

```diff
@@ -29,7 +29,11 @@
 namespace dev
 {
 
+/// Sets the data dir for the default ("ethereum") prefix.
+void setDataDir(std::string const& _dir);
 /// @returns the path for user data.
 std::string getDataDir(std::string _prefix = "ethereum");
+/// @returns the default path for user data, ignoring the one set by `setDataDir`.
+std::string getDefaultDataDir(std::string _prefix = "ethereum");
```

Comment:
This is required for backwards compatibility: On MacOS, if the data dir has not changed away from the default, the ipc socket is at a different place than the default data dir.